### PR TITLE
[oneDPL][RNG] Changing type of internal parameter of discard_block_engine

### PR DESCRIPTION
**Pull request by egrabovskaya**
Changing type of internal parameter from int to size_t. 
Note: changes break ABI backward compatibility.

### CODE REVIEW

**Review by aelizaro — DISMISSED** [review]
You can make PR name shorter and move a note about ABI changes to description

File: include/oneapi/dpl/internal/random_impl/discard_block_engine.h [hunk 1]
```diff
@@ -205,7 +205,7 @@ class discard_block_engine
     }
 
     _Engine engine_;
-    int n_ = 0;
+    size_t n_ = 0;
```
Comment (rarutyun) [inline include/oneapi/dpl/internal/random_impl/discard_block_engine.h:208]: It should be `::std::size_t` (actually, in every place of use) and it has missing include (e.g. `cstddef`). I am wondering how tests are passed. Seems like you don't include API headers first (before standard headers) or have some kind of umbrella header. Anyway, this header (and, probably, others) is not self-contained.

Comment (egrabovskaya) [reply include/oneapi/dpl/internal/random_impl/discard_block_engine.h]: "std" namespace for size_t and missing includes are added

**Comment by egrabovskaya** [comment]
We have two failed steps:
https://github.com/oneapi-src/oneDPL/actions/runs/3267430013/jobs/5372599082
https://github.com/oneapi-src/oneDPL/actions/runs/3267430013/jobs/5372599367

I suppose it's because the specified c++ version of the std compiler flag wasn't changed after this commit:
https://github.com/oneapi-src/oneDPL/commit/5a5b9534bd3446f0fb6a7f10d0499c2a2a6a4332
According to this, the failures are not related to the PR

**Review by paveldyakov — APPROVED** [review]

**Review by rarutyun — APPROVED** [review]

**Review by aelizaro — APPROVED** [review]

**Pull request merged**